rfcs/065-library-data-link-explorer/README.md

```diff
@@ -0,0 +1,129 @@
+
+Records may be linked implicitly by sharing common identifiers, but more often the linking data is manually added and managed by colleagues in collections information (CI). This usually takes the form of an identifier for one system being stored in a particular field in another. This can lead to rich and complex tangles of records where the merging logic sometimes leads to results that are unexpected or even non-deterministic.
+
+However, CI colleagues currently have no visibility over how records are linked other than by manually inspecting individual records. Since 2021, our team has maintained a script for visualising the way records are linked (we call these "matcher graphs" after the name of the pipeline service which identifies the links), but this script can only be run by software engineers due to the AWS permissions it requires. Since the source information belongs to Collections Information, we would like them to be able to view these graphs (nodes and edges) themselves.
+
+### How will it work?
+
+
+1. **User inputs a work ID**: A Collections Information staff member enters an work ID into the application.
+2. **API fetches the data**: The API queries Elasticsearch for records linked to the work ID.
+3. **Frontend displays the graph**: The frontend renders a visual graph of nodes (works) and edges (relationships), allowing users to explore the connections using a graph library.
+4. **Enhanced Collections Information workflow**: This allows Collections Information colleagues to independently view and explore relationships between records without requiring software engineer assistance, boosting productivity and collaboration.
+
+---
+
+## Backend
+
+- An API endpoint, powered by AWS Lambda to:
+    - query Elasticsearch for works, returning matched works of the given work id
+    - generate a DOT notation graph for this given work id using matcher logic within the endpoint
+    - transform this DOT notation into JSON ready for front end consumption
+- Accessed securely using AWS Secrets Manager for credentials for Elasticsearch
+
+### Helpers
+
+- Use the build of the content-api as an example API endpoint
+- JSON conversion is not strictly necessary, but useful for better graph libraries
+- [existing matcher logic](https://github.com/wellcomecollection/catalogue-pipeline/blob/main/pipeline/matcher_merger/matcher/scripts/getMatcherGraph.ts)
+
+### Technical Considerations
+
+- Frameworks and Dependencies
+    - TypeScript
+    - @elastic/elasticsearch client
+    - aws-lambda
+    - AWS SDK
+
+---
+
+## Frontend
+
+- A Next.js application that:
```

Comment:
Is the intention to host and deploy this using Vercel?

Reply:
For clarity, although we shouldn't be building new services we have to support using Vercel, as discussed for this project it's a good compromise to allow work to be released quickly and tested with end users. If this project moves to longer term support it can be ported to AWS. This is with the proviso that no secrets (e.g. Elasticsearch API keys) are kept in the Vercel platform.

Reply:
Agreed - will stick with Vercel as discussed, but may see as things progress to take it elsewhere